pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -881,28 +989,31 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundHTTPListenerOptsForPor
 	// single 0.0.0.0:port listener and use vhosts to distinguish
 	// individual http services in that port
 	if *currentListenerEntry, exists = listenerMap[*listenerMapKey]; exists {
-		// NOTE: This is not a conflict. This is simply filtering the
-		// services for a given listener explicitly.
-		if (*currentListenerEntry).locked {
-			return false, nil
-		}
-		if pluginParams.Service != nil {
-			if !(*currentListenerEntry).servicePort.Protocol.IsHTTP() {
-				outboundListenerConflict{
-					metric:          model.ProxyStatusConflictOutboundListenerTCPOverHTTP,
-					node:            pluginParams.Node,
-					listenerName:    *listenerMapKey,
-					currentServices: (*currentListenerEntry).services,
-					currentProtocol: (*currentListenerEntry).servicePort.Protocol,
-					newHostname:     pluginParams.Service.Hostname,
-					newProtocol:     pluginParams.Port.Protocol,
-				}.addMetric(pluginParams.Push)
+		// Skip conflict check if using protocol sniffing
```

Comment:
The code that you moved is *NOT* a conflict check. This piece of logic says that when the user declares their own ports in Sidecar.egress with some specific services on those ports, we should not generate any more listeners on that port as the user does not want those listeners.

```
// This is a catch all egress listener with no port. This
			// should be the last egress listener in the sidecar
			// Scope. Construct a listener for each service and service
			// port, if and only if this port was not specified in any of
			// the preceding listeners from the sidecarScope. This allows
			// users to specify a trimmed set of services for one or more
			// listeners and then add a catch all egress listener for all
			// other ports. Doing so allows people to restrict the set of
			// services exposed on one or more listeners, and avoid hard
			// port conflicts like tcp taking over http or http taking over
			// tcp, or simply specify that of all the listeners that Istio
			// generates, the user would like to have only specific sets of
			// services exposed on a particular listener.
			//
			// To ensure that we do not add anything to listeners we have
			// already generated, run through the outboundListenerEntry map and set
			// the locked bit to true.
			// buildSidecarOutboundListenerForPortOrUDS will not add/merge
			// any HTTP/TCP listener if there is already a outboundListenerEntry
			// with locked bit set to true
```

Reply:
Thanks for the explanation!